src/inputs/dnstap/DnstapInputStream.cpp

```diff
@@ -60,6 +56,24 @@ void DnstapInputStream::_read_frame_stream_file()
                 _logger->warn("dnstap data is wrong type or has no message, skipping frame of size {}", len_data);
                 continue;
             }
+            if (_filter_host) {
+                if (d.message().has_query_address()) {
```

Comment:
hm i think this logic needs some work. i believe dnstap will most often provide both the query _and_ the response address, and it should pass the filter if it matches either. if dnstap provides only one or the other, then it needs to just match that one i guess.

Reply:
changed the logic. please evaluate again.